app/src/main/java/de/rampro/activitydiary/ui/history/HistoryActivity.java

```diff
@@ -185,25 +202,37 @@ protected void onNewIntent(Intent intent) {
     private void handleIntent(Intent intent) {
         if (ActivityDiarySuggestionProvider.SEARCH_ACTIVITY.equals(intent.getAction())) {
             String query = intent.getStringExtra(SearchManager.QUERY);
+
             Uri data = intent.getData();
-            if(data != null) {
+            if (data != null) {
                 long id = Long.decode(data.getLastPathSegment());
                 filterHistoryView(id);
             }
-        }else if (ActivityDiarySuggestionProvider.SEARCH_NOTE.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_NOTE.equals(intent.getAction())) {
             Uri data = intent.getData();
-            if(data != null) {
+            if (data != null) {
                 String query = data.getLastPathSegment();
                 filterHistoryNotes(query);
             }
 
-        }else if (ActivityDiarySuggestionProvider.SEARCH_GLOBAL.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_GLOBAL.equals(intent.getAction())) {
             Uri data = intent.getData();
-            if(data != null) {
+            if (data != null) {
                 String query = data.getLastPathSegment();
                 filterHistoryView(query);
             }
-        }else if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_DATE.equals(intent.getAction())) {
+            Uri data = intent.getData();
+            if (data != null) {
+                String query = data.getLastPathSegment();
+                if (checkDateFormat(query, getDateFormat(query))) {
+                    filterHistoryDates(query);
+                } else{
+                    setWrongColorSearchText();
+                    Toast.makeText(getApplication().getBaseContext(), "Please enter date in format:\nyear.month.day or day.month.year", Toast.LENGTH_LONG).show();
```

Comment:
please don't use text directly, always go through string resources, otherwise we cannot translate it...